staging/src/k8s.io/apiextensions-apiserver/test/integration/testserver/resources.go

```diff
@@ -289,53 +304,50 @@ func checkForWatchCachePrimed(crd *apiextensionsv1beta1.CustomResourceDefinition
 			"spec":    map[string]interface{}{},
 		},
 	}
-	if _, err := resourceClient.Create(instance); err != nil {
-		return err
+	createdInstance, err := resourceClient.Create(instance)
+	if err != nil {
+		return false, err
+	}
+	err = resourceClient.Delete(createdInstance.GetName(), nil)
+	if err != nil {
+		return false, err
 	}
-	// we created something, clean it up
-	defer func() {
-		resourceClient.Delete(instanceName, nil)
-	}()
 
-	noxuWatch, err := resourceClient.Watch(metav1.ListOptions{ResourceVersion: initialListListMeta.GetResourceVersion()})
+	noxuWatch, err := resourceClient.Watch(metav1.ListOptions{ResourceVersion: createdInstance.GetResourceVersion()})
```

Comment:
isn't there a chance that we multiple delete events if this func return s false once? I would prefer having another wait loop here, waiting until the watch returns no error anymore.

Reply:
> I would prefer having another wait loop here, waiting until the watch returns no error anymore

if the watch cache initializes after `createdInstance.GetResourceVersion()`, it will never be able to establish a watch at that version (it will always report 'too old'). in that case, we have to go out and come back in again.

Reply:
> isn't there a chance that we multiple delete events if this func returns false once?

we always start from the resourceVersion of our created instance, which avoids delete events from any previous iterations. I'd take this as-is to clear up the flakes

Reply:
> we always start from the resourceVersion of our created instance

makes sense.